README.rst

```diff
@@ -109,7 +109,25 @@ languages.
 Specification
 =============
 
-[Describe the syntax and semantics of any new language feature.]
+The ``typing.sealed`` decorator can be applied to the definition of any class.
+This decoration indicates to type checkers that all immediate subclasses of the
+decorated class are defined in the current file.
+
+The exhaustiveness checking features of type checkers should assume that there
+are no subclasses outside the current file, treating the decorated class as a
+``Union`` of all its same-file subclasses.
+
+Type checkers should raise an error if a sealed class is inherited in a file
+different from where the sealed class is declared.
+
+A sealed class is automatically declared to be abstract. Whatever actions a
+type checker normally takes with abstract classes should be taken with sealed
+classes as well. What exactly these behaviors are (e.g. disallowing
+instantiation) is outside the scope of this PEP.
+
+Similar to the ``typing.final`` decorator, the only runtime behavior of this
```

Comment:
Let's add a link to the `Final` PEP: https://peps.python.org/pep-0591/

Reply:
This is actually not documented in that PEP or in the [Python documentation](https://docs.python.org/3/library/typing.html#typing.final). It is only mentioned in the [docstring of `final`](https://github.com/python/cpython/blob/main/Lib/typing.py#L2580-L2582). Which of these three things do you think should be linked?

Reply:
Oh, you want a link to support the lack of runtime enforcement, not a link for the runtime behavior of `__final__`. We don't actually have an sentence explicitly stating that. There probably should be. I am not sure if that should be in the spec section or a different section.

Reply:
I was actually just wanting a link to `typing.final` in general, so that if someone wanted to know what `typing.final` was, they could click a link and immediately start learning about it.

I wasn't thinking of trying to support any particular section in the sealed spec, just add a convenient link.

Reply:
I am thinking too much like an academic.